plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -355,8 +357,10 @@ public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
         catch (AmazonServiceException e) {
             throw new TrinoException(HIVE_METASTORE_ERROR, e);
         }
-        dropTableData(table.io(), table.operations().current());
-        deleteTableDirectory(fileSystemFactory.create(session), schemaTableName, table.location());
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+        BulkOperationsFileIO bulkOperationsFileIO = new BulkOperationsFileIO(fileSystem);
```

Comment:
Same for other catalogs. 
```suggestion
        BulkOperationsFileIo bulkOperationsFileIo = new BulkOperationsFileIO(fileSystem);
```